### PR TITLE
MGDOBR-1211 - Scaffold infra and manager v2 modules

### PR DESCRIPTION
**Pull request by r00ta**
[MGDOBR-1211](https://issues.redhat.com/browse/MGDOBR-1211))

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] The layers in the `kustomize` folder have been updated accordingly.
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

File: manager-parent/manager-v2/pom.xml [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        <filtering>true</filtering>
+      </testResource>
+    </testResources>
+
+    <plugins>
+      <plugin>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <version>${compiler-plugin.version}</version>
+        <configuration>
+          <forceJavacCompilerUse>true</forceJavacCompilerUse>
```
Comment (manstis) [inline manager-parent/manager-v2/pom.xml:44]: Damn it.. this is a left-over from my stuff.. it ensures Maven emits meaningful stack traces in the event of failing to compile (with Java 9). See https://stackoverflow.com/questions/46878649/maven-compilation-issue-with-java-9 (which shows the type of error I was getting locally).

**Review by manstis — APPROVED** [review]
You could remove the spurious `forceJavacCompilerUse` now or later if you like.

File: infra-parent/infra-v2/src/main/java/com/redhat/service/smartevents/infra/v2/api/V2APIConstants.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    /**
+     * Base Path for the Smart Events application API v2.
+     */
+    public static final String V2_ROOT = APIConstants.ROOT + "/v2";
+
+    /**
+     * Base Path for the Cloud Providers API
+     */
+    public static final String V1_CLOUD_PROVIDERS_BASE_PATH = V2_ROOT + "/cloud_providers";
```
Comment (sutaakar) [inline infra-parent/infra-v2/src/main/java/com/redhat/service/smartevents/infra/v2/api/V2APIConstants.java:18]:
```suggestion
    public static final String V2_CLOUD_PROVIDERS_BASE_PATH = V2_ROOT + "/cloud_providers";
```
Same with other constants below

**Review by sutaakar — APPROVED** [review]
Looks good, apart from the naming issue

**Pull request merged**